components/ws-daemon/pkg/content/archive.go

```diff
@@ -81,12 +81,12 @@ func BuildTarbal(ctx context.Context, src string, dst string, fullWorkspaceBacku
 	if err != nil {
 		return xerrors.Errorf("cannot open archive for writing: %w", err)
 	}
-	defer fout.Close()
-	defer func(e *error) {
+	defer func(e error) {
 		if e != nil {
 			os.Remove(dst)
 		}
-	}(&err)
+	}(err)
+	defer fout.Close()
```

Comment:
shouldn't we first close the file handle and the delete the file?

Reply:
The defer statements follow LIFO model. The first defer is executed at the last whilst the last defer statement is executed first.

Some examples here: https://www.geeksforgeeks.org/defer-keyword-in-golang/